### PR TITLE
Reinstated v2.1.0 converters

### PR DESCRIPTION
**Pull request by kunagpal**
Closes #70

### CODE REVIEW

File: lib/converters/v1.0.0/converter-v1-to-v21.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+     * mapper function, and creates the V2 auth parameter object.
+     *
+     * @param {Object} requestV1 - A Collection V1 compliant request instance
+     * @returns {{type: *}} - The v2.1.0 compliant request object
+     */
+    auth: function (requestV1) {
+        var result,
+            auth = Builders.super_.prototype.auth.call(this, requestV1);
+
+        if (!(auth && auth.type && auth[auth.type])) { return; }
```
Comment (shamasis) [inline lib/converters/v1.0.0/converter-v1-to-v21.js]: dont bail out on `auth[auth.type]`

File: lib/converters/v1.0.0/converter-v1-to-v21.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+        result = { type: auth.type }; // @todo: Replace this with ES6 style dynamic property definition later
+        result[auth.type] = _.map(auth[auth.type], function (value, key) {
```
Comment (shamasis) [inline lib/converters/v1.0.0/converter-v1-to-v21.js]: add a todo for handling other non underscore types.

File: lib/converters/v2.1.0/converter-v21-to-v1.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+     *
+     * @param {Item|Object} item - The wrapped request instance to be converted.
+     * @param {String} collectionId - The collection id to be added to the converted v1.0.0 compliant request.
+     * @param {Boolean} skipResponses - Will result in saved responses being ignored for conversion if set to true.
+     * @returns {Object} An equivalent v1.0.0 representation for the current v2.1.0 request.
+     */
+    request: function (item, collectionId, skipResponses) {
+        var auth = _.get(item, 'request.auth');
+
+        if (auth && auth.type && _.isArray(auth[auth.type])) {
```
Comment (shamasis) [inline lib/converters/v2.1.0/converter-v21-to-v1.js]: same bailout behaviour change

File: lib/converters/v2.1.0/converter-v21-to-v1.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            auth[auth.type] = _.transform(auth[auth.type], function (a, element) {
```
Comment (shamasis) [inline lib/converters/v2.1.0/converter-v21-to-v1.js]: add todo to later handle non `type` and non underscore parameters

File: test/schema/v1.0.0/converter-v1-to-v21.test.js [hunk 5]
```diff
@@ -69,7 +71,8 @@ describe('v1.0.0 ==> v2.0.0', function () {
 
                 result = validator.validate(converted, schema);
                 if (!result) {
-                    console.log(JSON.stringify(validator.error, null, 4)); // Helps debug on CI
+                    console.log(JSON.stringify(validator.error.subErrors, null, 4));
```
Comment (shamasis) [inline test/schema/v1.0.0/converter-v1-to-v21.test.js]: stray console log. do process.env.CI

File: lib/converters/v1.0.0/converter-v1-to-v21.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+            return {
+                key: key,
+                value: value,
+                type: typeMap[typeof value] || 'any'
```
Comment (shamasis) [inline lib/converters/v1.0.0/converter-v1-to-v21.js]: if a type is already present anywhere, respect that.

Comment (kunagpal) [reply lib/converters/v1.0.0/converter-v1-to-v21.js]: That is already being done, this is a v1 to v2.1 conversion. Type information is not available for v1 formats, so a derivation will have to be done here.

File: lib/converters/v1.0.0/converter-v1-to-v21.js (hunk 1 above)
Comment (kunagpal) [reply lib/converters/v1.0.0/converter-v1-to-v21.js]: Addressed

File: lib/converters/v1.0.0/converter-v1-to-v21.js (hunk 2 above)
Comment (kunagpal) [reply lib/converters/v1.0.0/converter-v1-to-v21.js]: Addressed

File: lib/converters/v1.0.0/converter-v1-to-v21.js (hunk 6 above)
Comment (kunagpal) [reply lib/converters/v1.0.0/converter-v1-to-v21.js]: Addressed

File: lib/converters/v2.1.0/converter-v21-to-v1.js (hunk 4 above)
Comment (kunagpal) [reply lib/converters/v2.1.0/converter-v21-to-v1.js]: Addressed

File: test/schema/v1.0.0/converter-v1-to-v21.test.js (hunk 5 above)
Comment (kunagpal) [reply test/schema/v1.0.0/converter-v1-to-v21.test.js]: Addressed

**Pull request merged**